paymentsheet/src/main/java/com/stripe/android/paymentsheet/paymentdatacollection/polling/PollingAuthenticator.kt

```diff
@@ -26,13 +31,36 @@ internal class PollingAuthenticator : PaymentAuthenticator<StripeIntent>() {
         authenticatable: StripeIntent,
         requestOptions: ApiRequest.Options
     ) {
-        val args = PollingContract.Args(
-            clientSecret = requireNotNull(authenticatable.clientSecret),
-            statusBarColor = host.statusBarColor,
-            timeLimitInSeconds = UPI_TIME_LIMIT_IN_SECONDS,
-            initialDelayInSeconds = UPI_INITIAL_DELAY_IN_SECONDS,
-            maxAttempts = UPI_MAX_ATTEMPTS,
-        )
+        val args = when (authenticatable.paymentMethod?.type) {
+            PaymentMethod.Type.Upi ->
+                PollingContract.Args(
+                    clientSecret = requireNotNull(authenticatable.clientSecret),
+                    statusBarColor = host.statusBarColor,
+                    timeLimitInSeconds = UPI_TIME_LIMIT_IN_SECONDS,
+                    initialDelayInSeconds = UPI_INITIAL_DELAY_IN_SECONDS,
+                    maxAttempts = UPI_MAX_ATTEMPTS,
+                    ctaText = R.string.stripe_upi_polling_message,
+                )
+            PaymentMethod.Type.Blik ->
+                PollingContract.Args(
+                    clientSecret = requireNotNull(authenticatable.clientSecret),
+                    statusBarColor = host.statusBarColor,
+                    timeLimitInSeconds = BLIK_TIME_LIMIT_IN_SECONDS,
+                    initialDelayInSeconds = BLIK_INITIAL_DELAY_IN_SECONDS,
+                    maxAttempts = BLIK_MAX_ATTEMPTS,
+                    ctaText = R.string.stripe_blik_confirm_payment,
+                )
+            else ->
+                // Just default to UPI
+                PollingContract.Args(
+                    clientSecret = requireNotNull(authenticatable.clientSecret),
+                    statusBarColor = host.statusBarColor,
+                    timeLimitInSeconds = UPI_TIME_LIMIT_IN_SECONDS,
+                    initialDelayInSeconds = UPI_INITIAL_DELAY_IN_SECONDS,
+                    maxAttempts = UPI_MAX_ATTEMPTS,
+                    ctaText = R.string.stripe_upi_polling_message,
+                )
```

Comment:
I’m fine with us just erroring here via `error("Received invalid payment method type ${authenticatable.paymentMethod?.type?.code} in PollingAuthenticator")`.